csm_web/scheduler/views/export.py

```diff
@@ -0,0 +1,590 @@
+import csv
+import datetime
+import io
+from typing import Generator, Iterable, List, Optional, Tuple
+
+from django.contrib.postgres.aggregates import ArrayAgg, JSONBAgg
+from django.core.exceptions import BadRequest
+from django.db.models import CharField, Count, Q, Value
+from django.db.models.functions import Concat
+from django.http.response import StreamingHttpResponse
+from rest_framework.decorators import api_view
+from scheduler.models import Attendance, Course, Section, Student
+
+
+@api_view(["GET"])
+def export_data(request):
+    """
+    Endpoint: /api/export
+
+    GET: Returns a CSV file of exported data.
+        Query parameters:
+            preview: int or None
+                if int > 0, then returns only that many entries from the database
+            courses: int[]
+                comma-separated list of course ids
+            fields: str[]
+                comma-separated list of fields
+            type: str
+                type of data to export
+    """
+
+    export_type = request.query_params.get("type", None)
+    courses_str = request.query_params.get("courses", None)
+    fields_str = request.query_params.get("fields", "")
+    preview = request.query_params.get("preview", None)
+
+    if courses_str is None or export_type is None:
+        raise BadRequest(
+            "Must include `courses` and `type` fields in the query parameters"
+        )
```

Comment:
Maybe make this a comprehendible error on the frontend if someone doesn't select any courses rather than defaulting to BadRequest? It could also not error and instead return a blank file.

Reply:
Reworked the frontend query to take care of this case individually.